### PR TITLE
Add elasticache user and group support

### PR DESCRIPTION
**Pull request by swhite-oreilly**
This PR includes two new modules for Elasticache [Users](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticache-user.html) and [User Groups](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticache-usergroup.html). 

Also including a filter update to ignore the default subnet for elasticache.

### Testing

Elasticache users and users groups were created. using the setup code mentioned below, and then AWS Nuke was used to clean these resources up, specifying  `ElasticacheUser` and `ElasticacheUserGroup`.  The resources were handle as expected:

```
...
us-east-1 - ElasticacheUserGroup - myusergroup - would remove
...
us-east-1 - ElasticacheSubnetGroup - default - Cannot delete default subnet group
us-east-1 - ElasticacheSubnetGroup - mygroup - would remove
...
us-east-1 - ElasticacheUser - default - Cannot delete default user
us-east-1 - ElasticacheUser - user1 - would remove
```

### Setup code

```
# Create a Redis cluster
echo "Creating Redis cluster"
aws elasticache create-cache-cluster \
--cache-cluster-id my-cluster \
--cache-node-type cache.t4g.micro \
--engine redis \
--num-cache-nodes 1 \
--cache-parameter-group default.redis7

# Create a Memcached cluster
echo "Creating Memcached cluster"
aws elasticache create-cache-cluster \
--cache-cluster-id my-memcached-cluster \
--cache-node-type cache.r4.large \
--engine memcached \
--engine-version 1.4.24 \
--cache-parameter-group default.memcached1.4 \
--num-cache-nodes 3

echo "Getting a subnet from the default VPC"
DEFAULT_VPC_ID=$(aws ec2 describe-vpcs --filters "Name=isDefault,Values=true" --query 'Vpcs[0].VpcId' --output text)

SUBNET_ID=$(aws ec2 describe-subnets --filters "Name=vpc-id,Values=$DEFAULT_VPC_ID" "Name=availability-zone,Values=us-east-1*" --query 'Subnets[0].SubnetId' --output text)
echo "Using subnet $SUBNET_ID"

# Create an elasticache subnet group (Note: This command will fail if the subnet does not exist)
echo "Creating elasticache subnet group"
aws elasticache create-cache-subnet-group \
    --cache-subnet-group-name "mygroup" \
    --cache-subnet-group-description "my subnet group" \
    --subnet-ids "$SUBNET_ID"

# Create a elasticache parameter group
echo "Creating elasticache parameter group"
aws elasticache create-cache-parameter-group \
    --cache-parameter-group-family "redis5.0" \
    --cache-parameter-group-name "mygroup" \
    --description "mygroup"

# Create a elasticache user
echo "Creating elasticache user"
aws elasticache create-user \
    --user-id user1 \
    --user-name myUser \
    --passwords PASSWORD_STRING \
    --engine redis \
    --access-string "on ~app::* -@all +@read"

# Create a elasticache user group
echo "Creating elasticache user group"
aws elasticache create-user-group \
    --user-group-id myusergroup \
    --engine redis \
    --user-ids default
```

### CODE REVIEW

File: resources/elasticache-users.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+func init() {
+	register("ElasticacheUser", ListElasticacheUsers)
+}
+
+func ListElasticacheUsers(sess *session.Session) ([]Resource, error) {
+	svc := elasticache.New(sess)
+
+	params := &elasticache.DescribeUsersInput{MaxRecords: aws.Int64(100)}
```
[the remark this answers is not in the document]
Comment (swhite-oreilly) [reply resources/elasticache-users.go]: @bjoernhaeuser paging added.

File: resources/elasticache-usergroups.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+func init() {
+	register("ElasticacheUserGroup", ListElasticacheUserGroups)
+}
+
+func ListElasticacheUserGroups(sess *session.Session) ([]Resource, error) {
+	svc := elasticache.New(sess)
+
+	params := &elasticache.DescribeUserGroupsInput{MaxRecords: aws.Int64(100)}
```
[the remark this answers is not in the document]
Comment (swhite-oreilly) [reply resources/elasticache-usergroups.go]: @bjoernhaeuser paging added.

File: resources/elasticache-usergroups.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (i *ElasticacheUserGroup) String() string {
+	return *i.groupId
+}
```
Comment (der-eismann) [inline resources/elasticache-usergroups.go:67]: Sorry for the late reply - would you mind adding properties for both resources as well?

Comment (der-eismann) [reply resources/elasticache-usergroups.go:74]: No these are fine, thanks!

**Review by der-eismann — APPROVED** [review]

**Pull request merged**